Test of "signac diff" relies on dictionary order

### Description
The test here relies on the order of a dictionary's items. On Python 3.7+, dictionaries are guaranteed to be ordered (and also on Python 3.6 even though it's not an API guarantee). This is not true on Python 3.5, so the test will occasionally fail.
https://github.com/glotzerlab/signac/blob/eb5cd298f65e99ff20f5df09e2a3fb0005e23c83/tests/test_shell.py#L264-L266

Instead, we could assert that each of the expected lines is in the output (and not care about their order). This would be a robust test that doesn't require us to change our API guarantees about diffs being ordered (which shouldn't be a requirement).

### To reproduce
Requires Python 3.5 and some luck.

### Error output
https://app.circleci.com/pipelines/github/glotzerlab/signac/891/workflows/5305ba03-9283-4faf-84f1-a20264be1f18/jobs/5768
```python
=================================== FAILURES ===================================

___________________________ TestBasicShell.test_diff ___________________________

self = <test_shell.TestBasicShell object at 0x7f92e012b6a0>
    def test_diff(self):
        self.call('python -m signac init ProjectA'.split())
        project = signac.Project()
        job_a = project.open_job({"a": 0, "b": 1})
        job_a.init()
        job_b = project.open_job({"a": 0, "b": 0})
        job_b.init()
        out = self.call('python -m signac diff {} {}'.format(job_a.id, job_b.id).split())
        expected = [str(job_a.id), "{'b': 1}", str(job_b.id), "{'b': 0}"]
>       assert out.strip().split(os.linesep) == expected
E       assert ['7f9fb369851...', "{'b': 1}"] == ['06b0c030eaf...', "{'b': 0}"]
E         At index 0 diff: '7f9fb369851609ce9cb91404549393f3' != 06b0c030eafd5424365f26898c9afeda'
E         Full diff:
E           [
E         +  '06b0c030eafd5424365f26898c9afeda',
E         +  "{'b': 1}",
E            '7f9fb369851609ce9cb91404549393f3',
E            "{'b': 0}",
E         -  '06b0c030eafd5424365f26898c9afeda',
E         -  "{'b': 1}",
E           ]

/home/circleci/repo/tests/test_shell.py:266: AssertionError
```
### System configuration

Please complete the following information:

 - Operating System [e.g. macOS]: Linux (CI)
 - Version of Python [e.g. 3.7]: 3.5
 - Version of signac [e.g. 1.0]: latest (1.4)

--- a/tests/test_shell.py
+++ b/tests/test_shell.py
@@ -263,7 +263,9 @@ class TestBasicShell():
         job_b.init()
         out = self.call('python -m signac diff {} {}'.format(job_a.id, job_b.id).split())
         expected = [str(job_a.id), "{'b': 1}", str(job_b.id), "{'b': 0}"]
-        assert out.strip().split(os.linesep) == expected
+        outputs = out.strip().split(os.linesep)
+        for line in expected:
+            assert line in outputs
 
     def test_clone(self):
         self.call('python -m signac init ProjectA'.split())

--- a/tests/test_shell.py
+++ b/tests/test_shell.py
@@ -264,8 +264,7 @@ class TestBasicShell():
         out = self.call('python -m signac diff {} {}'.format(job_a.id, job_b.id).split())
         expected = [str(job_a.id), "{'b': 1}", str(job_b.id), "{'b': 0}"]
         outputs = out.strip().split(os.linesep)
-        for line in expected:
-            assert line in outputs
+        assert set(expected) == set(outputs)
 
     def test_clone(self):
         self.call('python -m signac init ProjectA'.split())

